dali/kernels/signal/moving_mean_square_test.cc

```diff
@@ -0,0 +1,141 @@
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include "dali/test/tensor_test_utils.h"
+#include "dali/kernels/signal/moving_mean_square.h"
+
+namespace dali {
+namespace kernels {
+namespace test {
+
+namespace {
+const int kNDims = 1;
+}
+
+template<class InputType>
+class MovingMeanSquareCpuTest : public ::testing::Test {
+ public:
+  MovingMeanSquareCpuTest() {
+    input_.resize(dataset_size(shape_));
+  }
+
+
+  void SetUp() final {
+    std::mt19937_64 rng;
+    FillInput(rng);
+    calc_output();
+  }
+
+
+  std::vector<InputType> input_;
+  std::vector<float> ref_output_;
+  int window_size_ = 2048;
+  int buffer_length_ = 32000;
+  int reset_interval_ = 5000;
+  TensorShape<kNDims> shape_ = {buffer_length_};
+
+ private:
+  void calc_output() {
+    ref_output_.resize(buffer_length_);
+    for (int i = 0; i < buffer_length_; i++) {
+      float sumsq = 0;
+      for (int j = 0; j < window_size_ && i + j < buffer_length_; j++) {
+        auto val = static_cast<float>(input_[i + j]);
+        sumsq += val * val;
+      }
+      ref_output_[i] = sumsq / window_size_;
+    }
+  }
+
+
+  size_t dataset_size(const TensorShape<kNDims> &shape) {
+    return volume(shape);
+  }
+
+
+  template<typename RNG>
+  std::enable_if_t<std::is_signed<InputType>::value>
+  FillInput(RNG &rng) {
+    UniformRandomFill(input_, rng, -100, 100);
+  }
+
+
+  template<typename RNG>
+  std::enable_if_t<!std::is_signed<InputType>::value>
+  FillInput(RNG &rng) {
+    UniformRandomFill(input_, rng, 0, 100);
+  }
+};
+
+using TestTypes = ::testing::Types<uint8_t, uint16_t, int8_t, int16_t, int32_t, float>;
+
+TYPED_TEST_SUITE(MovingMeanSquareCpuTest, TestTypes);
+
+namespace {
+
+template<class GtestTypeParam>
+using TestedKernel = signal::MovingMeanSquareCpu<GtestTypeParam>;
+
+}  // namespace
+
+TYPED_TEST(MovingMeanSquareCpuTest, CalcSumSquaredTest) {
+  int st = 10, len = 200;
+  float ref = 0;
+  for (int i = st; i < len + st; i++) {
+    ref += this->input_[i] * this->input_[i];
+  }
+  EXPECT_FLOAT_EQ(signal::detail::CalcSumSquared(make_cspan(this->input_), st, len), ref);
+}
+
+
+TYPED_TEST(MovingMeanSquareCpuTest, CheckKernel) {
+  check_kernel<TestedKernel<TypeParam>>();
+  SUCCEED();
+}
+
+
+TYPED_TEST(MovingMeanSquareCpuTest, SetupTest) {
+  TestedKernel<TypeParam> kernel;
+  KernelContext ctx;
+  InTensorCPU<TypeParam, kNDims> in(this->input_.data(), this->shape_);
+  auto reqs = kernel.Setup(ctx, in, {this->window_size_});
+  ASSERT_EQ(this->shape_, reqs.output_shapes[0][0]) << "Kernel::Setup provides incorrect shape";
+}
+
+
+TYPED_TEST(MovingMeanSquareCpuTest, RunTest) {
+  TestedKernel<TypeParam> kernel;
+  KernelContext ctx;
+  InTensorCPU<TypeParam, kNDims> in(this->input_.data(), this->shape_);
+
+  auto reqs = kernel.Setup(ctx, in, {this->window_size_});
+
+  auto out_shape = reqs.output_shapes[0][0];
+  std::vector<float> output;
+  output.resize(dali::volume(out_shape));
```

Comment:
```suggestion
  output.resize(out_shape.num_elements())
```
Just a suggestion / information that the function is there.